### PR TITLE
Add option for plot

### PR DESCRIPTION
**Pull request by dh031200**
Pull Request for #1731 

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhanced customization for YOLO detection plotting and deprecated argument handling.

### 📊 Key Changes
- Added new arguments in prediction plot method, such as `conf`, `line_width`, `font_size`, `font`, `pil`, `img`, `labels`, `boxes`, `masks`, and `probs`.
- Replaced `hide_labels` and `hide_conf` with `show_labels` and `show_conf` across documentation, configurations, and code.
- Introduced a warning function to notify users about deprecated arguments and a mechanism to handle parameter deprecation.

### 🎯 Purpose & Impact
- The added arguments provide more flexibility in plotting object detections, offering a better user customization experience.
- Switching from a "hide" to "show" paradigm for labels and confidence information simplifies the user's understanding of display options.
- Informing users about deprecated arguments and their replacements helps transition to updated configurations, ensuring a smooth user experience and maintaining codebase clarity.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@dh031200 thanks for the PR! If we are going to introduce new arguments like hide_masks then they must be introduced uniformly across all ultralytics plotting locations, i.e. also in v8/detect/predict.py and then the terminology standardized in default.yaml. 

@AyushExel @Laughing-q what should we use for standard terminology here? `show_boxes=True` or `hide_boxes=False`?

**Comment by AyushExel** [comment]
@glenn-jocher how about just boxes=False, and its default value is true:
```
    def plot(self,
             show_conf=True,
             line_width=None,
             font_size=None,
             font='Arial.ttf',
             pil=False,
             example='abc',
             img=None,
             labels=True,
             boxes=True,
             masks=True):
```

**Comment by Laughing-q** [comment]
@glenn-jocher @AyushExel hey, how about we just add more customized args in results/plot function and remain the current plotting functionality in `predict.py`? 
Then the current plotting in predict.py is like a internal plotting which is fixed but out-of-the-box, and users can use the plot function with Results if they want more customization for plotting.
This case we don't need to add any extra args in our configs and we actually can add more customized args in Results/plot, i.e colors.

**Comment by AyushExel** [comment]
Yeah I think this PR is only adding functionality to results/plot.py

**Comment by AyushExel** [comment]
@glenn-jocher take a look when you're ready

**Comment by glenn-jocher** [comment]
@AyushExel hey buddy, you've got modifications to YOLOv5 tutorials in here. Let's revert those first ok?

**Comment by glenn-jocher** [comment]
@AyushExel remember all YOLOv5 tutorials now live in the ultralytics repo and are published to https://docs.ultralytics.com/yolov5 after last week's major PR (i.e. +3000 lines). So anytime we make updates to YOLOv8 we want to leave YOLOv5 alone.

**Comment by glenn-jocher** [comment]
@AyushExel ah also deprication -> deprecation

**Comment by AyushExel** [comment]
reverted v5 tutorials

**Comment by glenn-jocher** [comment]
I've reverted the tutorial notebook changes, not sure how they got in there but I don't think this PR should alter the notebook right?

**Comment by glenn-jocher** [comment]
@AyushExel I see something odd is going on with the classification/prediction logits. We don't plot any `logits` ever, these are the logistic values like -10 or +10 that we convert to probabilities, or `probs` of i.e. 0.01 or 0.9999 etc.

I removed this wording in the past, so I think you may have started your changes an out of date branch. You should always make sure that you start any changes from the very latest main branch.

**Comment by AyushExel** [comment]
Okay, I'll investigate the logit plotting. Yeah, notebook changes are not needed in this PR. There's something else too but I'm not sure where it was introduced. On running test_results() test, new files are created in the assets folder which didn't happen last I worked on the results.py file.
Screenshot 2023-04-02 at 9 07 46 PM

**Comment by AyushExel** [comment]
@glenn-jocher the branch is up to date. The problem is here. It's in master https://github.com/ultralytics/ultralytics/blob/main/ultralytics/yolo/engine/results.py#L68

**Comment by AyushExel** [comment]
Okay logits fixed
Screenshot 2023-04-02 at 9 10 53 PM

**Comment by AyushExel** [comment]
> @AyushExel I see something odd is going on with the classification/prediction logits. We don't plot any `logits` ever, these are the logistic values like -10 or +10 that we convert to probabilities, or `probs` of i.e. 0.01 or 0.9999 etc.
> 
> I removed this wording in the past, so I think you may have started your changes an out of date branch. You should always make sure that you start any changes from the very latest main branch.

yeah, logit is used for classification model outputs. Generally non-normalized values but it is limited to classification so tell a bit more than `probs`. But I think we need a better name

**Comment by glenn-jocher** [comment]
> Okay, I'll investigate the logit plotting. Yeah, notebook changes are not needed in this PR. There's something else too but I'm not sure where it was introduced. On running test_results() test, new files are created in the assets folder which didn't happen last I worked on the results.py file. Screenshot 2023-04-02 at 9 07 46 PM

Yes this is expected. Tests are creating single channel and PNG with alpha images to run inference on.

**Comment by glenn-jocher** [comment]
> Okay logits fixed Screenshot 2023-04-02 at 9 10 53 PM

Wait the cls plotting should be in a vertical list like what's produced by `yolo predict yolov8n-cls.pt`. What I was saying about the logits is that your image is not showing logits, it's showing probabilities. Logits are what the model outputs directly, and what certain loss function can operate directly on, i.e. BCEWithLogitsLoss(), They are unbounded and can range from -inf to inf, but are usually in a range of say -10 to 10 with 3-sigma confidence, i.e. 99% of the logit values will be hanging around the range of -10 to 10. 

After you pass logits through a sigmoid in the YOLO head, then these values are no longer called logits, they're usually called "probabilities". So our annotation functions should never mention logits as we never plot them or show them to users in any form or in any prediction output, i.e. there are no logits in any Result object in YOLO.

```python
probability = sigmoid(logit)
```

**Comment by glenn-jocher** [comment]
@AyushExel I think one of these might be incorrect since they should both match up? Also please update deprication -> deprecation in the PR, thanks!
Screenshot 2023-04-03 at 00 57 52

**Comment by AyushExel** [comment]
@glenn-jocher this is the code to reproduce plotting. I'm getting the same probabilities.
```
    model = YOLO('yolov8n-cls.pt')
    res = model(SOURCE)
    plt = res[0].plot()
    cv2.imshow("plt", plt)
    cv2.waitKey(0)
```

**Comment by glenn-jocher** [comment]
@dh031200 @AyushExel PR merged into updates branch for 8.0.62 release soon. Thank you for your contributions!
